lib/persisted-model.js

```diff
@@ -1060,19 +1073,25 @@ PersistedModel.bulkUpdate = function(updates, callback) {
   var Model = this;
   var idName = this.dataSource.idName(this.modelName);
   var Change = this.getChangeModel();
+  var conflicts = [];
 
   updates.forEach(function(update) {
+    var id = update.change.modelId;
     switch (update.type) {
       case Change.UPDATE:
+        tasks.push(function(cb) {
+          applyUpdate(Model, id, update.data, update.change, conflicts, cb);
+        });
+        break;
+
       case Change.CREATE:
-        // var model = new Model(update.data);
-        // tasks.push(model.save.bind(model));
         tasks.push(function(cb) {
-          var model = new Model(update.data);
-          model.save(cb);
+          applyCreate(Model, id, update.data, update.change, conflicts, cb);
         });
         break;
       case Change.DELETE:
+        // NOTE(bajtos) Delete does not care whether the model has
+        // the same revision as expected, we are going to delete it anyway
```

Comment:
> Delete does not care whether the model has the same revision as expected

The delete should be a conflict if the current target revision doesn't match the previous revision of the source. If someone has changed the target, we shouldn't blindly delete it (although I'm not certain that is what you are doing).